From d2e13a34b89abd3e4be4ccca4aa7364d8d83473d Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Wed, 25 Nov 2015 16:48:11 +0800
Subject: [PATCH] MA-7253 bcmdhd: avoid null interface before update tx status

Avoid to access the null interface before update tx status.

This patch workaround to fix the panic when the p2p connection
disconnect in the p2p stress test. The interface will be
released during the disconnect process.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 drivers/net/wireless/bcmdhd/dhd_linux.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/bcmdhd/dhd_linux.c b/drivers/net/wireless/bcmdhd/dhd_linux.c
index 37a786a..26c3710 100644
--- a/drivers/net/wireless/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/bcmdhd/dhd_linux.c
@@ -2147,6 +2147,9 @@ dhd_txcomplete(dhd_pub_t *dhdp, void *txp, bool success)
 #ifdef PROP_TXSTATUS
 	if (dhdp->wlfc_state && (dhdp->proptxstatus_mode != WLFC_FCMODE_NONE)) {
 		dhd_if_t *ifp = dhd->iflist[DHD_PKTTAG_IF(PKTTAG(txp))];
+		/* if the interface is released somewhere just return */
+		if (ifp == NULL)
+			return;
 		uint datalen  = PKTLEN(dhd->pub.osh, txp);
 
 		if (success) {
-- 
1.8.0

